tree-sitter-darklang/grammar.js

```diff
@@ -442,11 +442,44 @@ module.exports = grammar({
       repeat1(
         choice(
           // higher precedence than escape sequences
-          token.immediate(prec(1, /[^\\"\n]+/)),
+          token.immediate(prec(1, /[^\\"]+/)),
           $.char_or_string_escape_sequence,
         ),
       ),
 
+    // $"hello {name}
+    string_interpolation: $ =>
+      seq(
+        field("symbol_dollar_sign", alias("$", $.symbol)),
+        field("symbol_open_quote", alias('"', $.symbol)),
+        optional(
+          field("string_interpolation_content", $.string_interpolation_content),
+        ),
+        field("symbol_close_quote", alias('"', $.symbol)),
+      ),
+
+    string_interpolation_content: $ =>
+      repeat1(choice($.string_to_eval, $.string_text)),
+
+    string_text: $ =>
+      prec.left(
+        repeat1(
+          choice(
+            token.immediate(prec(1, /[^\\"{}\n]/)),
+            $.char_or_string_escape_sequence,
+          ),
+        ),
+      ),
+
+    string_to_eval: $ =>
+      seq(
+        field("symbol_open_brace", alias("{", $.symbol)),
+        field("expr", $.expression),
+        field("symbol_close_brace", alias("}", $.symbol)),
+      ),
+
+    string_segment: $ => choice($.string_literal, $.string_interpolation),
```

Comment:
Hmm, maybe `string_expr` would be a better name, since `string_segment` is already a concept

Reply:
in PT we call it `StringSegment` should we rename that as well?

Reply:
Hm, actually this is right - the thing I was thrown off by was that `string_segment` was right next to `int_literal` and such - was expecting _that_ to be something like `string_expr`.